### PR TITLE
Small improvements to heading parameter docs

### PR DESCRIPTION
**Pull request by Andrew15-5**
- Added a few missing `{}` wrappers for values/fields;
- changed "one" to "`{1}`";
- changed "behaviour" to "behavior".

CC @tingerrr

### CODE REVIEW

File: crates/typst/src/model/heading.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ use crate::util::{option_eq, NonZeroExt};
 /// can be set to configure the starting depth.
 #[elem(Locatable, Synthesize, Count, Show, ShowSet, LocalName, Refable, Outlinable)]
 pub struct HeadingElem {
-    /// The absolute nesting depth of the heading, starting from one. If set
+    /// The absolute nesting depth of the heading, starting from `{1}`. If set
```
Comment (laurmaedje) [inline crates/typst/src/model/heading.rs:49]: why this change?

Comment (reknih) [reply crates/typst/src/model/heading.rs:49]: I think this actually reduces legibility. Going from words to numbers makes your brain do a context-switch. This is why it's good style in many languages to spell out small numbers, e.g. up to 12.

Comment (Andrew15-5) [reply crates/typst/src/model/heading.rs:49]: Really? Well, I guess we hit the subjectivity ceiling.

File: crates/typst/src/model/heading.rs [hunk 2]
```diff
@@ -65,18 +65,18 @@ pub struct HeadingElem {
...
-    /// The relative nesting depth of the heading, starting from one. This is
+    /// The relative nesting depth of the heading, starting from `{1}`. This is
     /// combined with `{offset}` to compute the actual `{level}`.
     ///
     /// This is set by the heading syntax, such that `[== Heading]` creates a
-    /// heading with logical depth 2, but actual level `{offset + 2}`. If you
-    /// construct a heading manually, you should typically prefer this over
-    /// setting the absolute `level`.
+    /// heading with logical `{depth}` of `{2}`, but the actual `{level}` is
+    /// `{offset + 2}`. If you construct a heading manually, you should
```
Comment (reknih) [inline crates/typst/src/model/heading.rs:73]: `offset + 2` is the important highlight. Previously, it stood out, now it drowns in a sea of other highlights. It is true that the other words you emphasized are also properties, but this is not about using them but rather about understanding the concepts behind it. The change of typeface should always be considered as it disrupts the reading flow.

File: crates/typst/src/model/heading.rs [hunk 3]
```diff
@@ -65,18 +65,18 @@ pub struct HeadingElem {
...
+    /// typically prefer this over setting the absolute `{level}`.
     #[default(NonZeroUsize::ONE)]
     pub depth: NonZeroUsize,
 
-    /// The starting offset of each heading's level, used to turn its relative
-    /// `{depth}` into its absolute `{level}`.
+    /// The starting offset of each heading's `{level}`, used to turn its
```
Comment (reknih) [inline crates/typst/src/model/heading.rs:78]: Same here

File: crates/typst/src/model/heading.rs (hunk 2 above)
Comment (Andrew15-5) [reply crates/typst/src/model/heading.rs:73]: I don't understand what you mean by that. Do you want me to completely revert the changes and only leave the "behavior" fix? Or do some other changes also can stay?

Comment (Andrew15-5) [reply crates/typst/src/model/heading.rs:73]: I also wanted to added syntax highlighting here:
image

since it already has backticks in place. Will those changes be ok to do?

Comment (laurmaedje) [reply crates/typst/src/model/heading.rs:73]: Behavior fix is okay and "logical depth of 2" is also an improvement.

Since the `shift` changes are unrelated, let's do them after the release.

**Comment by laurmaedje** [comment]
Thanks!
